cmd/zhack/zhack.c

```diff
@@ -469,6 +479,331 @@ zhack_do_feature(int argc, char **argv)
+	spa_t *spa = NULL;
+	dsl_pool_t *dp = NULL;
+	dsl_scan_t *scn = NULL;
+	int c;
+	uint64_t physmem_override = 0;
+
+	// Scan the entire DDT
+	zfs_scrub_ddt_class_max = DDT_CLASS_UNIQUE;
+
+	// Disable reference tracking debugging
+	reference_tracking_enable = B_FALSE;
+
+	// Disable prefetch during scan
+	zfs_no_scrub_prefetch = B_TRUE;
+
+	while ((c = getopt(argc, argv, "D:EG:H:PRTi:S:nrv")) != -1) {
+		switch (c) {
+		case 'D':
+			// How much of the DDT are we scanning?
+		{
+			char *endptr = NULL;
+			unsigned long class =
+			    strtoul(optarg, &endptr, 0);
+			if ((errno == 0) && (*endptr == '\0') &&
+			    (class < DDT_CLASSES)) {
+				zfs_scrub_ddt_class_max = class;
+			} else {
+				fatal(NULL, FTAG, "DDT class must be between "
+				    "0 and %d, inclusive", DDT_CLASSES-1);
+			}
+		}
+		break;
+		case 'E':
+			// Forcibly reset DDT class max after import
+			do_ddt_reset++;
+			break;
+		case 'G':
+		{
+			char *endptr = NULL;
+			uint64_t gap = strtoul(optarg, &endptr, 0);
+			if ((errno == 0) && (*endptr == '\0')) {
+				zfs_scan_max_ext_gap = gap;
+			} else {
+				fatal(NULL, FTAG, "Bad range tree gap (-G)");
+			}
+		}
+		break;
+		case 'H':
+		{
+			char *endptr = NULL;
+			uint64_t fact = strtoul(optarg, &endptr, 0);
+			if ((errno == 0) && (*endptr == '\0') &&
+			    (fact >= 1) && (fact <= 1000)) {
+				zfs_scan_mem_lim_fact = fact;
+			} else {
+				fatal(NULL, FTAG, "Bad hard factor (-H)");
+			}
+		}
+		break;
+		case 'M':
+		{
+			char *endptr = NULL;
+			uint64_t mem = strtoul(optarg, &endptr, 0);
+			if ((errno == 0) && (*endptr == '\0') &&
+			    (mem >= (1 << 15)) &&
+			    (mem <= sysconf(_SC_PHYS_PAGES))) {
+				physmem_override = mem;
+			} else {
+				fatal(NULL, FTAG,
+				    "Bad physical memory override (-M)");
+			}
+		}
+		break;
+		case 'P':
+			// Turn prefetching back on
+			zfs_no_scrub_prefetch = B_FALSE;
+			break;
+		case 'R':
+			// Restart a scrub
+			do_restart++;
+			break;
+		case 'S':
+		{
+			char *endptr = NULL;
+			uint64_t fact = strtoul(optarg, &endptr, 0);
+			if ((errno == 0) && (*endptr == '\0') &&
+			    (fact >= 1) &&
+			    (fact <= 1000)) {
+				zfs_scan_mem_lim_soft_fact = fact;
+			} else {
+				fatal(NULL, FTAG, "Bad soft factor (-S)");
+			}
+		}
+		break;
+		case 'T':
+			// Turn reference tracking back on
+			reference_tracking_enable = B_TRUE;
+			break;
+		case 'i':
+		{
+			char *endptr = NULL;
+			uint64_t intval = strtoul(optarg, &endptr, 0);
+			if ((errno == 0) && (*endptr == '\0')) {
+				zfs_scan_checkpoint_intval = intval;
+			} else {
+				fatal(NULL, FTAG, "Bad scan interval (-i)");
+			}
+		}
+		break;
+		case 'n':
+			// Don't launch a scrub, just resume one
+			no_spawn++;
+			break;
+#if 0
+		// This function is too horrifying to contemplate calling.
+		case 'o':
+			if (set_global_var(optarg) != 0)
+				usage();
+			break;
+#endif
+		case 'r':
+			// Resilver instead of scrub
+			do_resilver++;
+			break;
+		case 'v':
+			// Be chatty
```

Comment:
[nit] I think 'verbose' is self-explanatory enough that we don't need the comment. In general I'd drop the comment for most of the one-line just-increment options, since their behavior should be documented in the usage and the man page, but that's my preference.